tests/integration/dbapi/async/V2/test_queries_async.py

```diff
@@ -417,3 +419,41 @@ async def test_bytea_roundtrip(
         assert (
             bytes_data.decode("utf-8") == data
         ), "Invalid bytea data returned after roundtrip"
+
+
+@fixture
+async def setup_db(connection_system_engine_no_db: Connection, use_db_name: str):
+    use_db_name = use_db_name + "_async"
+    with connection_system_engine_no_db.cursor() as cursor:
+        await cursor.execute(f"CREATE DATABASE {use_db_name}")
+        yield
+        await cursor.execute(f"DROP DATABASE {use_db_name}")
+
+
+@mark.xfail("dev" not in environ[API_ENDPOINT_ENV], reason="Only works on dev")
+async def test_use_database(
```

Comment:
Is it possible to reuse the code between v1 and v2 even for tests? The functions `test_use_database()` look very similar.

Reply:
While it is possible, some of the fixtures those tests use are V1/V2 specific. If we move this test up above the v1/v2 directories and call the shared code from inside it will create an indirection that's really hard to follow, since it's evaluated at runtime. I'd rather have duplication in test code than obfuscation.
![image](https://github.com/firebolt-db/firebolt-python-sdk/assets/93913847/4e9bd288-dcd9-4e31-a716-793969537a60)